Axis

eric-nieuwland opened the pull request:
After the update to allow minimum and maximum dates I noticed the (date) axes were sometimes a bit off. On closer inspection I saw the same glitch in the showcase `detailed.svg`.
- Created some unit tests that express my expectations
- Fixed the code so the unit tests succeed

The axes now always align on computed major unit boundaries. So the code in XAxis that attempted to compensate for misalignment is no longer needed.

alex-rowley commented:
Hi Eric,

Thank you for this PR. Overall it looks like amazing work.  

One of the things that I was trying to do with detailed.svg was to have the lines of the line chart actually touching the y axis. As a result the gap between the vertical grid lines and the y-axis is slightly smaller than the gap between the first and second vertical grid lines. Whereas what we have now (if I run pytest locally) is that there is a small gap between where the lines of the linechart start and the y-axis is and all of the vertical grid lines are evenly spaced including with the y-axis. 

The difference between the two is, imo, largely down to aesthetic style. Both are valid, one avoids the inconsistency of not having grid lines evenly spaced, the other avoids the awkward gap in the line before the chart really begins. 

Broadly, i'd be ok with keeping either or both, (maybe both?) but maybe let the user pick.

Just wondering what your thoughts were on this.

Thanks

Alex

eric-nieuwland commented:
Hi Alex,


It is a joy to contribute bits and pieces to the great work you're doing.

IMHO it is - just like you said - a matter of aesthetics.
I am a sucker for visual consistency and regularity, so I like (and must admit: expect) the major grid to align with the axes.
But I don’t want to force my preference upon others, so we may allow users to choose:
align to major grid
align to minor grid (need to define that precisely)
use an arbitrary value, with a default coming from the data points

Btw, if your goal is to have the graph touch the y axis, why not just add some data points? ;-)


—eric

> On 14 May 2025, at 09:03, Alex Rowley ***@***.***> wrote:
> 
> 
> alex-rowley
>  left a comment 
> (alex-rowley/py-svg-chart#6)
>  <https://github.com/alex-rowley/py-svg-chart/pull/6#issuecomment-2878947542>
> Hi Eric,
> 
> Thank you for this PR. Overall it looks like amazing work.
> 
> One of the things that I was trying to do with detailed.svg was to have the lines of the line chart actually touching the y axis. As a result the gap between the vertical grid lines and the y-axis is slightly smaller than the gap between the first and second vertical grid lines. Whereas what we have now (if I run pytest locally) is that there is a small gap between where the lines of the linechart start and the y-axis is and all of the vertical grid lines are evenly spaced including with the y-axis.
> 
> The difference between the two is, imo, largely down to aesthetic style. Both are valid, one avoids the inconsistency of not having grid lines evenly spaced, the other avoids the awkward gap in the line before the chart really begins.
> 
> Broadly, i'd be ok with keeping either or both, (maybe both?) but maybe let the user pick.
> 
> Just wondering what your thoughts were on this.
> 
> Thanks
> 
> Alex
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/alex-rowley/py-svg-chart/pull/6#issuecomment-2878947542>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/ACQ43R5F5YIN7ZRTCUFG6ZD26LTDPAVCNFSM6AAAAAB4WNQBGGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDQNZYHE2DONJUGI>.
> You are receiving this because you authored the thread.
>

eric-nieuwland commented:
Updated so you can now choose how to shift:
- False: do not shift
- True: automatic shift to align right/bottom of graph with x-/y-axis
- <value>: shift left/down by this amount

Changed `test_pysvgchart.py` to show off shifting.

Dropped support for Python versions before 3.10 because of the type hints used.

alex-rowley commented:
Thanks, merged as part of 382e9c055f54 with a minor tweak or two.
